src/OrchardCore.Modules/OrchardCore.Media/wwwroot/Scripts/media.js

```diff
@@ -3351,7 +3356,7 @@ Vue.component('mediaFieldThumbsContainer', {
                         <div class="media-container-main-item-title card-body">\
                             <a href="javascript:;" class="btn btn-light btn-sm float-right inline-media-button delete-button"\
                                 v-on:click.stop="selectAndDeleteMedia(media)"><i class="fa fa-trash" aria-hidden="true"></i></a>\
-                            <span class="media-filename card-text small text-danger" :title="media.name">{{ media.name }}</span>\
+                            <span class="media-filename card-text small text-danger" :title="media.name">{{ media.attachedFileName == null ? media.name : media.attachedFileName }}</span>\
```

Comment:
I agree with @sebastienros here, we shouldn't be showing anything different in the edit UI here.

Reply:
I reverted my changes to the admin display in https://github.com/OrchardCMS/OrchardCore/pull/10570/commits/5cd1a8151491adf11b7c83334e50b226c6bae8cf